samcli/commands/test_runner/init/cli.py

```diff
@@ -0,0 +1,199 @@
+        tag_value=tag_value,
+        template_name=template_name,
+        env_file=env_file,
+        runtime=runtime,
+        image_uri=image_uri,
+    )
+
+
+def do_cli(
+    ctx: Any,
+    tag_key: Optional[str],
+    tag_value: Optional[str],
+    template_name: str,
+    env_file: str,
+    runtime: str,
+    image_uri: str,
+) -> None:
+    """
+    implementation of the `sam test-runner init` command
+    """
+
+    # TODO: Merge these two options into one to follow AWS CLI (if we can get access to their parser)
+    if tag_key and not tag_value:
+        LOG.info(COLOR.red("Specified tag key without a tag value."))
+        return
+
+    if tag_value and not tag_key:
+        LOG.info(COLOR.red("Specified tag value without a tag key."))
+        return
+
+    if tag_value and tag_key:
+        tag_filters = [{"Key": tag_key, "Values": [tag_value]}]
+    else:
+        tag_filters = None
+
+    # TODO: When ready, public ECR repositories will be created holding images capable of running tests for each available runtime.
+    #       Until ready for public images, image URIs are supplied by option
+    # image_uri = get_image_uri(runtime)
+
+    if tag_filters is not None:
+        boto_client_provider = get_boto_client_provider_with_config(region=ctx.region, profile=ctx.profile)
+        resource_arn_list = query_tagging_api(tag_filters, boto_client_provider)
+
+        if not resource_arn_list:
+            raise NoResourcesMatchGivenTagException(
+                f"Given tags `Key={tag_key},Value={tag_value}` do not match any resources, were they entered incorrectly?"
+            )
+
+        template_generator = FargateRunnerCFNTemplateGenerator(resource_arn_list)
+        arn_map_generator = FargateRunnerArnMapGenerator()
+
+        resource_arn_map_yaml_string = arn_map_generator.generate_env_vars_yaml_string(resource_arn_list)
+
+        write_file(env_file, resource_arn_map_yaml_string)
+
+        LOG.info(
+            COLOR.green(
+                f"✓ Sucessfully generated an environment variable specification file `test_runner_arn_map.yaml`.\n"
+            )
+        )
+        LOG.info(
+            COLOR.yellow(
+                "These environment variables hold the ARNs of your testable resources, and passing this file to `sam test_runner run` will make them available in the container running your tests.\n\n"
+                "Feel free to change them, or add new variables, but keep in mind that the test runner reserves the keys TEST_RUNNER_BUCKET, TEST_COMMAND_OPTIONS, and TEST_RUN_ID, and will throw an Exception if they are specified.\n\n"
+                "If you add new resources or change existing ones, you can use `sam test-runner get-arns` to generate an updated environment variable specification file.\n"
+            )
+        )
+    else:
+        # If no tags were provided we cannot pass resource ARNs to the template generator to generate IAM statements
+        # We also cannot generate a resource ARN map
```

Comment:
Should we still generate an empty mapping file so customers can add their own env variables?

Reply:
Maybe not an empty file, but definitely a message reminder saying "if you'd like to pass environment variables, create a YAML file etc."

I feel automatically creating empty YAML files might be more annoying for those who don't want to include environment variables than creating one will be for those who do